diff -ur grub-2.02~beta2/grub-core/Makefile.am grub-2.02~beta2-yosemite/grub-core/Makefile.am
--- grub-2.02~beta2/grub-core/Makefile.am	2013-12-24 17:40:31.000000000 +0100
+++ grub-2.02~beta2-yosemite/grub-core/Makefile.am	2015-04-22 18:00:02.000000000 +0200
@@ -55,6 +55,7 @@
 
 rs_decoder.h: $(srcdir)/lib/reed_solomon.c
 	$(TARGET_CC) $(TARGET_CPPFLAGS) $(TARGET_CFLAGS) -Os -I$(top_builddir) -S -DSTANDALONE -o $@ $< -g0 -mregparm=3 -ffreestanding
+	gsed -i 's/\(\.macos.*\)/\/* \1 *\//g' $@ || true
 
 CLEANFILES += grub_script.yy.c grub_script.yy.h
 
